lib/cdo/log_object.rb

```diff
@@ -0,0 +1,85 @@
+# LogObject is a simple container to save errors and important info
+# when executing a process. It can also time block execution and log the result.
+#
+# LogObject is helpful when we want to
+# - Prevent non-fatal errors from stopping process execution but still want to know about them.
+# - Bubble up combined errors and info from lower levels to higher levels.
+#   Even to external components such as HoneyBadger or Slack channel.
+
+class LogObject
+  attr_reader :errors, :logs
+
+  def initialize
+    @errors = []  # List of rescued error objects
+    @logs = []    # List of message logs
+  end
+
+  # Execute a block and time it.
+  # Save exception if caught, do not re-raise. Caller's flow will continue as normal.
+  #
+  # @param action_name [string] a friendly name of the block being executed
+  def time(action_name = nil)
+    start_time = Time.now
+    return unless block_given?
+
+    yield
+
+    info("#{action_name || 'Unnamed'} action completed without error in"\
+      " #{Time.now - start_time} seconds."
```

Comment:
Just looking at the sample output in Slack, it seems it might be a bit more readable if we don't show so many digits.  e.g. "in 2.53992749 seconds" is more difficult to read than just "in 2.53 seconds".

Reply:
will fix